__tests__/find-ruby.test.ts

```diff
@@ -1,74 +1,67 @@
-import * as io from '@actions/io';
+import * as tc from '@actions/tool-cache';
+import * as core from '@actions/core';
 import fs = require('fs');
 import os = require('os');
 import path = require('path');
+import * as cache from '../src/cache';
+import {run} from '../src/main';
 
-const toolDir = path.join(__dirname, 'runner', 'tools');
-const tempDir = path.join(__dirname, 'runner', 'temp');
+describe('find-ruby', () => {
+  let inSpy: jest.SpyInstance;
+  let tcSpy: jest.SpyInstance;
+  let cnSpy: jest.SpyInstance;
 
-process.env['AGENT_TOOLSDIRECTORY'] = toolDir;
-process.env['RUNNER_TOOL_CACHE'] = toolDir;
-process.env['RUNNER_TEMP'] = tempDir;
+  //beforeAll(async () => {});
 
-import {findRubyVersion} from '../src/installer';
+  beforeEach(() => {
+    tcSpy = jest.spyOn(tc, 'find');
+    inSpy = jest.spyOn(core, 'getInput');
+    cnSpy = jest.spyOn(process.stdout, 'write');
+    cnSpy.mockImplementation(line => {
+      // uncomment to debug
+      // process.stderr.write('write2:' + line + '\n');
+    });
+  });
 
-describe('find-ruby', () => {
-  beforeAll(async () => {
-    await io.rmRF(toolDir);
-    await io.rmRF(tempDir);
+  afterEach(() => {
+    tcSpy.mockClear();
+    cnSpy.mockClear();
+    jest.clearAllMocks();
```

Comment:
Whats the difference between mockClear() and clearAllMocks() do we need both?

If we do , we should also consider adding inSpy.mockClear(). It may be a no-op for now, but it may save us some troubleshooting down the line.